Mail/Views/Thread/Message/Attachment/AttachmentsView.swift

```diff
@@ -114,34 +143,54 @@ struct AttachmentsView: View {
             DocumentPicker(pickerType: .exportContent(attachmentsURL.urls))
                 .ignoresSafeArea()
         }
+        .onDisappear {
+            for (_, task) in trackDownloadTask {
+                task.cancel()
+            }
+        }
     }
 
     private func openAttachment(_ attachment: Attachment) {
+        isDownloadDisabled = true
         matomo.track(eventWithCategory: .attachmentActions, name: "open")
         previewedAttachment = attachment
         if !FileManager.default.fileExists(atPath: attachment.getLocalURL(mailboxManager: mailboxManager).path) {
-            Task {
-                await mailboxManager.saveAttachmentLocally(attachment: attachment)
+            downloadProgressState[attachment.uuid] = 0.0
+            trackDownloadTask[attachment.uuid] = Task { @MainActor in
+                await mailboxManager.saveAttachmentLocally(attachment: attachment) { progress in
+                    Task { @MainActor in
+                        downloadProgressState[attachment.uuid] = progress
+                    }
+                }
+                downloadProgressState[attachment.uuid] = 1.0
             }
         }
+        isDownloadDisabled = false
     }
 
     private func downloadSwissTransferAttachment(stUuid: String, fileUuid: String) {
+        isDownloadDisabled = true
         matomo.track(eventWithCategory: .attachmentActions, name: "openSwissTransfer")
 
-        Task {
+        trackDownloadTask[fileUuid] = Task {
             await tryOrDisplayError {
                 let attachmentURL = try await mailboxManager.apiFetcher.downloadSwissTransferAttachment(
-                    stUuid: stUuid,
-                    fileUuid: fileUuid
-                )
+                    stUuid: stUuid, fileUuid: fileUuid
+                ) { progress in
+                    Task { @MainActor in
+                        downloadProgressState[fileUuid] = progress
+                    }
+                }
                 attachmentsURL = AttachmentsURL(urls: [attachmentURL])
             }
         }
+        isDownloadDisabled = false
     }
 
     private func downloadAllAttachments() {
         downloadInProgress = true
+        isDownloadDisabled = true
+        isShowingProgressCircle = true
         Task {
```

Comment:
downloadAll isn't tracked so it can't be cancelled.

We can assign a random uuid for this task since we don't have something from the API.
```suggestion
      trackDownloadTask[UUID().stringValue] =  Task {
```